### PR TITLE
Fix: add some missing modifiers from gems

### LINKED ISSUE #2887: List of all Gem Qualities that are not calculated / don't work
Currently, quite a few alternate (and Default) Qualities from Gems don't work / aren't implemented. Some of these have no way to be represented in PoB right now or simply don't matter for any Calculations, but some DO matter and were overlooked it seems.
Some of these have been reported over the Months, but far from all I think.

So here is the list of Gems that have Quality on them that should seemingly work because the Functions already properly work on other Gems or Items (e.g. Reservation Reductions or Stat Requirement Reductions) or because there are dedicated Checkboxes tied to the Function of the Quality available in the Config. E.g. Predator Support makes "Is the enemy marked with Signal Prey" appear, so a check exists and the Gem not adding its Quality Benefits seems like an oversight or a Typo in the Code? Idk.

- Blasphemy Divergent (Alternate2) - Reservation missing
- Blastchain Mine Anomalous (Alternate1) - Reservation missing
- Chance to Bleed Divergent (Alternate2) - Damage missing
- Clarity Divergent (Alternate2) - Damage missing (with a hypothetical "on full Mana" Setting)
- Close Combat Anomalous + Divergent (Alternate1 and Alternate2) - Damage missing
- Controlled Destruction Divergent (Alternate2) - Damage missing
- Cruelty Default - Damage missing(but probably the Cruelty calc isn't known?)
- Dread Banner Anomalous (Alternate1) - Less Damage taken missing
- Efficacy Divergent (Alternate2) - Cost Reduction missing
- Empower Anomalous (Alternate1) - Stat Requirement Reduction not working
- Enhance Anomalous (Alternate1) - Stat Requirement Reduction not working
- Enlighten Anomalous (Alternate1) - Stat Requirement Reduction not working
- Essence Drain Anomalous (Alternate1) - Life Regen missing
- Freezing Pulse Anomalous (Alternate1) - Damage missing in certain Scenarios (Far Shot scaling)
- Frost Shield Phantasmal (Alternate3) - Life Regen missing
- Generosity Divergent (Alternate2) - Reservation missing
- Greater Multiple Projectiles Divergent (Alternate2) - Cost Reduction missing
- Immortal Call Divergent (Alternate2) - Results in less Elemental EHP and more Physical EHP
- Increased Duration Anomalous (Alternate1) - Cost Reduction missing
- Infernal Legion Divergent (Alternate2) - Damage missing by Minions
- Lancing Steel Divergent (Alternate2) - Damage missing in certain Scenarios (Point Blank scaling)
- Lesser Multiple Projectiles Anomalous (Alternate1) - Cost Reduction missing
- Meat Shield Default - Lower Minion Physical EHP
- Minion Damage Anomalous (Alternate1) - Results in less Damage dealt by Minions
- Pierce Divergent (Alternate2) - Damage missing (if pierced)
- Poachers Mark Anomalous (Alternate1) - Damage missing
- Precision Anomalous (Alternate1) - Reservation missing
- Predator Anomalous + Divergent (Alternate1 + Alternate2)  - Minus Allres or Crit Chance and Multi Missing
- Purifying Flame Phantasmal (Alternate3) - Results in lower Consecrated Ground Effect (Life Regen + Crit Chance)
- Slower Projectiles Anomalous (Alternate1) - Damage missing in certain Scenarios (Point Blank scaling)
- Storm Burst Divergent (Alternate2) - Too much AoE
- Trinity Anomalous (Alternate1) - Attack and Cast Speed missing
- Vortex Phantasmal (Alternate3) - Damage missing when cast on Frostbolt
- Warlords Mark Anomalous (Alternate1) - Damage taken on Enemies missing
- Wither Anomalous (Alternate1) - Chaos EHP increase missing
- Wrath Divergent (Alternate2) - Reservation missing
- Zealotry Anomalous (Alternate1) - Crit Multi missing

It would be awesome to one Day see these all working properly in PoB :)

-------------------------------------------------------------------------------------------------

And for Completions' Sake, here is the List of Qualities that aren't / can't be implemented / don't really matter for PoB as it is right now anyways / can't be expressed properly. 
Maybe these can serve as Inspiration for people to implement certain functionalities:
<details>
  <summary>Long</summary>

- Ancestral Protector Divergent
- Ancestral Warchief Divergent
- Animate Weapon Anomalous + Divergent 
- Arc Anomalous
- Arcane Cloak Anomalous
- Arcane Surge Divergent
- Arcanist Brand Phantasmal
- Arctic Armour Anomalous
- Armageddon Brand Divergent
- Assassin's Mark Default
- Ballista Totem Divergent + Phantasmal
- Bane Divergent
- Barrage Support Divergent
- Blade Vortex Divergent
- Bladefall Divergent
- Bladestorm Anomalous + Phantasmal
- Blast Rain Anomalous
- Blastchain Mine Divergent
- Blind Anomalous
- Blink Arrow Phantasmal
- Block Chance Reduction Default + Anomalous
- Blood Rage Anomalous
- Bloodlust Anomalous + Divergent
- Bodyswap Phantasmal
- Brand Recall Default
- Cast when Damage Taken Divergent
- Cast when Stunned Divergent
- Chain Anomalous + Divergent
- Chain Hook Divergent
- Chance to Flee Default
- Charged Dash Anomalous + Divergent
- Charged Mines Anomalous
- Charged Traps Anomalous + Divergent
- Cobra Lash Anomalous
- Consecrated Path Divergent
- Contagion Divergent
- Conversion Trap Anomalous
- Convocation Divergent 
- Corrupting Fever Divergent
- Crackling Lance Divergent
- Creeping Frost Phantasmal 
- Cruelty Divergent
- Culling Strike Divergent
- Dark Pact Divergent
- Dash Anomalous + Divergent
- Desecrate Divergent
- Despair Divergent
- Detonate Mines Default + Anomalous
- Discharge Divergent
- Divine Ire Anomalous
- Domainating Blow Phantasmal
- Double Strike Divergent
- Earthshatter Anomalous
- Enfeeble Anomalous
- Explosive Arrow Divergent
- Exsanguinate Divergent
- Faster Projectiles Anomalous
- Feeding Frenzy Default + Anomalous
- Fire Penetration Divergent
- Fireball Phantasmal
- Flame Dash Anomalous
- Flicker Strike Default + Divergent
- Fork Anomalous
- Freezing Pulse Divergent
- Frenzy Divergent
- Frost Shield Divergent
- Frostblink Divergent
- Frostbolt Anomalous + Phantasmal
- Galvanic Arrow Divergent
- General's Cry Divergent
- Greater Multiple Projectiles Divergent
- Greater Volley Anomalous
- Ground Slam Divergent
- Herald of Purity Anomalous
- Hexblast Phantasmal
- Ice Bite Divergent
- Ice Spear Phantasmal
- Impale Divergent
- Impending Doom Divergent
- Increased Critical Strikes Divergent
- Infernal Blow Anomalous
- Infernal Cry Phantasmal
- Innervation Anomalous + Divergent
- Inspiration Default
- Intensify Divergent + Phantasmal
- Item Rarity Anomalous (Meme)
- Lancing Steal Anomalous
- Lesser Multiple Projectiles Divergent
- Lifetap Divergent
- Lightning Arrow Phantasmal
- Lightning Spire Trap Divergent
- Lightning Trap Divergent
- Maim Anomalous
- Melee Splash Divergent
- Minefield Anomalous
- Mirror Arrow Phantasmal
- Multiple Traps Divergent
- Onslaught Anomalous
- Orb of Storms Anomalous
- Phase Run Divergent
- Pinpoint Anomalous + Divergent
- Plague Bearer Anomalous + Divergent 
- Poachers Mark Default
- Point Blank Divergent
- Portal Anomalous (Meme)
- Power Charge on Critical Anomalous
- Puncture Divergent + Phantasmal
- Punishment Anomalous
- Purifying Flame Anomalous
- Rage Phantasmal
- Rain of Arrows Anomalous + Phantasmal
- Raise Spectre Divergent + Phantasmal
- Raise Zombie Phantasmal
- Reap Divergent
- Reave Phantasmal (functionality is already there though)
- Riposte Divergent
- Rolling Magma Phantasmal
- Scorching Ray Default
- Scourge Arrow Anomalous
- Second Wind Divergent
- Seismic Cry Divergent
- Sigil of Power Anomalous
- Siphoninh Trap Divergent
- Slower Projectiles Divergent
- Spell Totem Divergent
- Splitting Steel Anomalous
- Storm Brand Divergent
- Storm Call Phantasmal
- Stun Anomalous + Divergent
- Summon Phantasm Default 
- Summon Raging Spirits Phantasmal
- Summon Skeletons Anomalous + Divergent + Phantasmal
- Sunder Anomalous
- Swift Affliction Divergent
- Temporal Chains Anomalous
- Unearth Divergent + Phantasmal
- Unleach Anomalous + Divergent
- Urgent Orders Divergent
- Vengeance Default
- Volley Anomalous 
- Vulnerability Divergent
- Warlords Mark Default
- Wild Strike Phantasmal
- Wither Divergent
- Withering Step Divergent
- Withering Touch Divergent
- 
</details>

### PR DESCRIPTION
**Pull request by Nostrademous**
Fixes #2887 

- Clarity Divergent (added configuration option for "FullMana")
- Predator Anomalous
- Dread Banner Anomalous (support for less damage to nearby enemies from quality)
- Empower Anomalous requirement reduction
- Enhance Anomalous requirement reduction
- Slower Projectile Anomalous increase to Nearby Enemies

Might add more

### CODE REVIEW

File: src/Data/Skills/sup_dex.lua [hunk 1]
```diff
@@ -3419,7 +3422,10 @@ skills["SupportSlowerProjectiles"] = {
 		},
 		["projectiles_damage_+%_to_nearby_targets"] = {
 			mod("Damage", "INC", nil, ModFlag.Projectile)
-		}
+		},
```
Comment (Wires77) [inline src/Data/Skills/sup_dex.lua:3425]: This appears to be the same stat as the previous one in the statMap, the wording just got changed.  I also don't think it should be dependent on nearby enemies, as most users won't realize that is what needs to change to make the damage visible.

File: src/Data/Skills/sup_int.lua [hunk 2]
```diff
@@ -1644,6 +1644,12 @@ skills["MinionFocusFire"] = {
 	skillTypes = { [SkillType.Spell] = true, [SkillType.Instant] = true, [SkillType.Minion] = true, [SkillType.Triggerable] = true, [SkillType.Duration] = true, [SkillType.InstantShiftAttackForLeftMouse] = true, [SkillType.Cooldown] = true, },
 	statDescriptionScope = "debuff_skill_stat_descriptions",
 	castTime = 0,
+	statMap = {
+		["resist_all_%"] = {
+			mod("EnemyModifier", "LIST", { mod = mod("ElementalResist", "BASE", nil) }),
+			mod("EnemyModifier", "LIST", { mod = mod("ChaosResist", "BASE", nil) }),
+		},
```
Comment (Wires77) [inline src/Data/Skills/sup_int.lua:1651]: This doesn't seem to do anything from my testing...

**Review by Wires77 — CHANGES_REQUESTED** [review]
The exceptional gems and predator support changes didn't seem to do anything from my test, and I think slower projectiles just needed a stat name change.  Here's a build with all of them for confirmation:
```bash
eNrVW1tz4jYUft7-Cg_vCZhLLh3SDiEkm5mQpZDdbZ92hH0ANbJFbTlZ-ut7JBlwCIZjdR9aHhJjn-uno3NRnO6v3yPhvUCSchlf1fzTRs2DOJAhj-dXtc9PtycXtV9_-ak7YmrxaXadcaGf_PLTh6659gS8gEC-mqdYMgf1ZS2p9Q0lLVmsFiDjIftTJncyvKo9yhhq3pTFIVfrb4FgafrIIriqTQJkrnksDSAO-9v7ljBiPJ7I4BnUXSKzpVH7wuF1KEOkuR-OPo2fCkp5XFSKNn_ojgRbQTJRTHkp_riq9dB1NoePXKEoJjKU0-40L08v27X6fo7JEiDcEPunreKnjGmUwGA2g0DxF-gnXPUXLA5gI-WsjK8q7TATii8Fh6RgYaeM40kqJm5Gk63sRgtdbx6ml1uoGmWUX7laXAsEqoL0-3nMFRQZWueN9unFRavdPO9c-GeHlL1nvjxvI3ODwjySPJWxAxAOxvYzIXAHFTlLUezLaMrjtyAS3eolwD7NbMyNWcizdCuhjGfIYtaXaWEnlFr2wGfwhrSUcjCh0Y1xF9IotZkjSDBDKBqDNrYSQ65hAoHELFRFRxWWgh9uyhw4BxMXBgdFE7XNQM3yRYe_ioR-pzRV3cB3mrwioe_7pckmVjR5RcKD9r1IZWrXMWRM7hh8HG1zzfnpeaPV8dvti2ancVGKqEEfixqPmBiy7zzKIixbT-wZ4kISKs_2OmoKCe5AfQt-1rT3cUALxc9xAikkL4WyeEZkyINri3DjIOMY5hDTjHoACBZ32DCMmQLaZtxQdQ6Do2lJ4GjCPeCcNNsdGssuPCfty4PK3uLTap4ezPUVIfrKkpCQMGJI5qvJgoMgUBs8iywkXIsMxJAoslT0e_DC0uLOLi-61htLTgsQwB4JGULYafL88kZS_qnbSFGNrZdEMkuI62GJSQ6MFquUB9gLsQjr9xjCLKBlwU03PJQvEGGMm7YaG_njvlwLHAF2HD_kkBCVOHpKseD5RoZzqKSkOsckWy5xp-toofLd8gRRTnmhBJ-cEag_4ZjTZ0tKy6m3FFXBlpqs4IHPFyrGDpiuZYeF7suCybSCM1tysopb7OZJrfxIvqLIhZ6R02rUWOq3Gb3UjgTiv1dk-W_ISQoGcZglOkbJOnY53qvp1s0Zgr66j5YyUeZmn4kgNSLv42WmvNgcAKQL-ap7H51bplIKYPFVTSUZ5Orf0D5zIb7FWTTVY7D9vc0nRcqIp8G3aTab6ROEGhqemGOPwe3toP90_2WQs0zAZDUvkEKwZQrhVW3GRIocHC8nWt0EM3mgSPTYtNl0SaLGqhskHAGnEOvZnGSyOcOgUOpDBRIhriATQAfsabUEHS0pFbM8RVKozckDidJO_jQHzQkDbRkgYCsS5aYVIVEPBPS40NWShtoQw9sWWho9lr6ETzNFjDYzUJDs0M01ycFi80jcHTTCvAEiGZH3exTavLrQghKbEPKy3cAMMNyJ296kEzsJ2sS6TqJds9dSL8XsegdRer3CanWr4d05XYqYwkCE6EGf5z5JnbNZoCB5sOe7ubI3YvL864UwY5nQ93_LmOBqhRN1q3g7l4FjtqfTuG57rBydA1Jcm4cH-6QnVC5CK1lrtQBoP8wBcM-0jea78dKcAvM4EFmII2tekDcWCzbVumveXJ8Y92UWY2mKudCn3GwqNJDWix3RWqpW_KGLpuS0d0JOmWhuZW_9KhL4a5GmDt3rhGc9nghd1rcdPBbRv6y7mgidhyTGicTPOUcsUca5OUSaYAiKhUyx-r1CdOoaovoWy3LhRQvDN7ZpDVsMAwsOXuniqCcBtNxI9Ypi60RYWkRUemHIdWwzYeLkR2JSJholNF0RGWCngpAcgMGKIweHbtLxgchv2SZHuzMBq7uAx43dU1XA0IR4tdWywcA5KPZarNcziiTmkzGw5RqgrjXgP7uFCw7smFN4kt8cM46Jd_KMnZ7SbWiF7a-BWTOm77X-uGV-p-rAau_d74bf22_rOwjqPzpDWj23MshSHWSluYC0-2-AqUXEkudiJvRJMIwSQLnyfxLEZJR78VxPRPsijRBbb7mpEWW5KuFoYHmPpLntjqMmKkXSyqbiOMoSSBWAuGZxvIOoW8nKIb5JgIU7QveW731IG2Yv5_4_Ae5fHgO8LxjOoav9QDdJ0O6KsE0qAdY14w6i-ZVpsp8SAI9Z3w2Pnzet-KX4LoO2saGRwuBJVvlkprGNZagb4c7FRQuNUiiu8B6Ff56_ImHZLaQD7GwgfNR8eRb-PH4wFx8WSi3Tn-v119fX0yWmQDmD79i5nQYyqi9RDFp5YpA90YrqPfxcz3vmYwTV15K69l2LdO05emMc1-7qi0eJk6J-pm-uvxgwvnB49VJgSbCYqER7-LeU0R_Ge93of8TUPGTLTQbTTx82baMZEhRTNxwRSswYtF4gTfj7-tCpa5Y5B77YMWUp2L-WfcVWBMuhvl2YJXLSA2RmqlonXuyB1-dFltDzJ69s6fWmqzRlwrMoeR3kUjYOG-vKuId1l61JYztMdS2lSh0N2ufLEaOuQaiKLLcYwM9ei2RSc59JLTd4aSA0fxBb283IMzdtZ-5ousTTkRX4CCICVVXRnkg6c9FzJPywsLzokbi64Jb7LvJd9LXdYsF3i7z28eWhAFtV7L7wcnSclHc6DvY45MDWMTXhyrMHkRWXbz_KTvHVcWEibUnKQjj52XGGtemw7mduO6nlFr6t6lHWdk_7DjHtO4PfogREm0LkYPa_aH6IK-m-CmcOUem-DP8iXKgpWfMSdw2d1AG5MY_ntE6gcqIohdcxXdDX08WdFgEn330Z2s5odVy8oewX9yA_YlIvysSe2tat51OkmYjNIGpeXpDxjM_fvWegh0zzlwh9tmJfNTz2EkO0eZP_Eefo6WoQQ6RfAzj4PkO4PvLcnH3aE5l96rr1tbHd-u5_d_wDVzHt_Q==
```

File: src/Data/Skills/sup_int.lua (hunk 2 above)
Comment (Nostrademous) [reply src/Data/Skills/sup_int.lua:1651]: weird... This 
```
	["nearby enemies have (%-%d+)%% to all resistances"] = function(num) return {
		mod("EnemyModifier", "LIST", { mod = mod("ElementalResist", "BASE", num) }),
		mod("EnemyModifier", "LIST", { mod = mod("ChaosResist", "BASE", num) }),
	} end,
	```
	works in supporting `Crown of the Tyrant` unique. Not sure why it doesn't work for the gem

Comment (Wires77) [reply src/Data/Skills/sup_int.lua:1651]: After messing with an "enemy on consecrated ground" mod, I determined that `EnemyModifier` doesn't work for mods from skills.  You can fake it with `GlobalEffect` and `Debuff`, but that might cause other side effects I'm not sure about.

Comment (Nostrademous) [reply src/Data/Skills/sup_int.lua:1651]: I reached the same conclusion

**Pull request merged**